From 57abdf44cb4480b996243315e804ac18cd7bcb19 Mon Sep 17 00:00:00 2001
From: Jean-Eudes Onfray <je@onfray.fr>
Date: Fri, 24 Feb 2012 14:17:01 +0100
Subject: [PATCH 02/03] Core: Enable alpha-blending with OpenGL for
 ocpnDC::DrawLines(...)

---
 src/ocpndc.cpp |   46 +++++-----------------------------------------
 1 file changed, 5 insertions(+), 41 deletions(-)

diff --git a/src/ocpndc.cpp b/src/ocpndc.cpp
index 674eb05..4c96552 100644
--- a/src/ocpndc.cpp
+++ b/src/ocpndc.cpp
@@ -350,48 +350,12 @@ void ocpnDC::DrawLines( int n, wxPoint points[], wxCoord xoffset, wxCoord yoffse
 
            glPushAttrib(GL_COLOR_BUFFER_BIT | GL_LINE_BIT | GL_HINT_BIT);      //Save state
 
-           bool b_draw_thick = false;
+           glEnable(GL_LINE_SMOOTH);
+           glEnable(GL_BLEND);
+           glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+           glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
 
-                  //      Enable anti-aliased lines, at best quality
-           if(b_hiqual)
-           {
-                 glEnable(GL_LINE_SMOOTH);
-                 glEnable(GL_BLEND);
-                 glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
-                 glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
-
-                 if(m_pen.GetWidth() > 1)
-                 {
-                       GLint parms[2];
-                       glGetIntegerv(GL_SMOOTH_LINE_WIDTH_RANGE, &parms[0]);
-                       if(m_pen.GetWidth() > parms[1])
-                             b_draw_thick = true;
-                       else
-                             glLineWidth(m_pen.GetWidth());
-                 }
-                 else
-                       glLineWidth(1);
-           }
-           else
-           {
-                 glDisable(GL_LINE_SMOOTH);
-                 glDisable(GL_BLEND);
-
-                 if(m_pen.GetWidth() > 1)
-                 {
-                       GLint parms[2];
-                       glGetIntegerv(GL_ALIASED_LINE_WIDTH_RANGE, &parms[0]);
-                       if(m_pen.GetWidth() > parms[1])
-                             b_draw_thick = true;
-                       else
-                             glLineWidth(m_pen.GetWidth());
-                 }
-                 else
-                       glLineWidth(1);
-           }
-
-
-           if(b_draw_thick/*m_pen.GetWidth() > 1*/)
+           if(m_pen.GetWidth() > 1)
            {
               wxPoint p0 = points[0];
               for(int i=1; i<n; i++)
-- 
1.7.10.2

